mne/export/tests/test_export.py

```diff
@@ -46,6 +48,31 @@ def test_export_raw_eeglab(tmpdir):
     assert_allclose(raw.get_data(), raw_read.get_data())
 
 
+@pytest.mark.skipif(not _check_edflib_installed(strict=False),
+                    reason='edflib-python not installed')
+@pytest.mark.parametrize('format', ['edf', 'bdf'])
+def test_export_raw_edf_and_bdf(tmpdir, format):
+    """Test saving a Raw instance to EDF format."""
+    fname = (Path(__file__).parent.parent.parent /
+             "io" / "tests" / "data" / "test_raw.fif")
+    raw = read_raw_fif(fname)
+
+    # only test with EEG channels
+    raw.pick_types(eeg=True, eog=True, ecg=True, emg=True)
+
+    raw.load_data()
+    temp_fname = op.join(str(tmpdir), f'test.{format}')
+    raw.export(temp_fname)
+    raw.drop_channels([ch for ch in ['epoc']
+                       if ch in raw.ch_names])
+    raw_read = read_raw_edf(temp_fname, preload=True)
+    assert raw.ch_names == raw_read.ch_names
+    print(len(raw.times), len(raw_read.times))
+    assert_array_almost_equal(raw.times, raw_read.times, decimal=1)
+    assert_array_almost_equal(
+        raw.get_data(), raw_read.get_data(), decimal=3)
```

Comment:
I am not sure to be honest... I suspect something to do with the added zero padding to the end to create a full data record when writing?

@Teuniz do you have any intuition why this would occur?

Reply:
Well, for starters, never use floating point arithmetic for for exact timestamps.  
For example, EDFbrowser and EDFlib use (64-bit) integers for timestamps in order to avoid rounding errors and   comparison errors. Also, a very tiny rounding error can easily accumulate when repeated and added every sample point.  
But to answer your question, I don't believe it's a problem with EDFlib.